src/Parsers.jl

```diff
@@ -765,7 +792,23 @@ end
     end
 end
 
-function checkdelim!(buf, pos, len, options::Options)
+function checkquote(source, pos, len, q::PtrLen)
+    eof(source, pos, len) && return pos
+    return checkdelim(source, pos, len, q)  # calls `incr!` if `q` matches.
+end
+
+function checkquote(source, pos, len, q::UInt8)
+    eof(source, pos, len) && return pos
+    b = peekbyte(source, pos)
+    if b == q
+        pos += 1
+        incr!(source)
+    end
+    return pos
+end
+
+# Used by CSV.jl
+function checkdelim!(buf::AbstractVector{UInt8}, pos, len, options::Options)
```

Comment:
added comment and type annotation here because i ended up being led astray by this function... and then wondered why we had it at all 😂